nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -173,16 +177,18 @@ def init_model(self, cfg: DictConfig, trainer: Trainer):
         if self.first_stage_of_pipeline() and self.virtual_prompt_style in [
             VirtualPromptStyle.P_TUNING,
         ]:
-
-            self.word_embeddings = self.frozen_model.model.language_model.embedding.word_embeddings
+            if self.frozen_model.mcore_gpt:
+                self.word_embeddings = self.frozen_model.model.embedding.word_embeddings
+            else:
+                self.word_embeddings = self.frozen_model.model.language_model.embedding.word_embeddings
 
         self.padded_vocab_size = self.frozen_model.padded_vocab_size
 
         # Prepare pseudo token ids for virtual/virtual prompt tokens
         self.pseudo_tokens = get_pseudo_tokens(self.max_virtual_tokens)
         if isinstance(self.tokenizer, SentencePieceTokenizer):
             if not self.tokenizer.legacy:
-                self.tokenizer.pad_token = self.tokenizer.ids_to_tokens([self.tokenizer.pad_id])[0]
+                self.tokenizer.pad_token = self.tokenizer.ids_to_tokens([self.tokenizer.eos_id])[0]
```

Comment:
llama2 tokenizer does not have a pad token, so we need to turn it into an eos token for [launcher script -> prompt learning] to work.
@ericharper should we rollback this change?

Reply:
+ @aklife97 to review.

Reply:
but this change is not model-specific. this would likely be an issue.
let's have a check that uses pad_id if it is present in the tokenizer. If not, we can use eos_id

Reply:
fixed.